react/BindingChallenge.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from 'react'
+import { ButtonWithIcon, IconClose } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import { useDevice } from 'vtex.device-detector'
+import { useCssHandles } from 'vtex.css-handles'
+
+import { useBinding } from './hooks/useBindings'
+import BindingSelectorBlock from './BindingSelectorBlock'
+
+const close = <IconClose />
+
+const CSS_HANDLES = ['challengeBar', 'actionCTA', 'actionContainer'] as const
+
+interface Props {
+  barText: string
+}
+
+const BindingChallenge = ({ barText }: Props) => {
+  const [show, setDisplay] = useState(false)
+  const [viewerCountry, setViewerCountry] = useState<string | null>(null)
+  const handles = useCssHandles(CSS_HANDLES)
+  const { device } = useDevice()
+
+  const {
+    data: { currentBinding, bindingsError, loadingBindings },
+  } = useBinding()
+
+  useEffect(() => {
+    fetch('/_v/binding-selector/viewer-country').then((response) =>
+      response.text().then((countryISO: string | null) => {
+        !countryISO || setViewerCountry(countryISO)
+      })
+    )
```

Comment:
Not sure how you feel about that, but I would prefer keeping the same style we are resolving promises throughout the code - with `async / await`. I know the pattern inside `useEffect` can be weird, so just want to hear your thoughts.

Reply:
Changed